packages/manager/modules/billing/src/autoRenew/actions/terminate-all-dom/controller.js

```diff
@@ -0,0 +1,130 @@
+import get from 'lodash/get';
+import set from 'lodash/set';
+import kebabCase from 'lodash/kebabCase';
+import map from 'lodash/map';
+import filter from 'lodash/filter';
+import { BillingService } from '@ovh-ux/manager-models';
+
+const DOMAIN_SERVICE_TYPE = 'DOMAIN';
+export default class {
+  /* @ngInject */
+  constructor($translate, $http, $q, atInternet, BillingAutoRenew) {
+    this.$translate = $translate;
+    this.$http = $http;
+    this.$q = $q;
+    this.atInternet = atInternet;
+    this.BillingAutoRenew = BillingAutoRenew;
+  }
+
+  $onInit() {
+    this.isLoading = false;
+    this.cancelAllDom = true;
+  }
+
+  selectAllDomains() {
+    this.domains.forEach((domain) => set(domain, 'selected', true));
+  }
+
+  cancelServices() {
+    const domainsSelected = filter(this.domains, (domain) => domain.selected);
+    if (this.cancelAllDom || domainsSelected.length > 0) {
+      this.isLoading = true;
+      const servicesToTerminate = [];
+      if (this.cancelAllDom) {
+        servicesToTerminate.push(
+          this.$http
+            .get(`/allDom/${this.serviceId}/serviceInfos`)
+            .then(({ data }) => {
+              return new BillingService({
+                ...data,
+                serviceId: data.id,
+                serviceType: this.serviceType,
+              });
+            }),
+        );
+        this.trackCancel(this.serviceType);
+      }
+      domainsSelected.forEach((domain) => {
+        servicesToTerminate.push(
+          this.$http
+            .get(`/domain/${domain.name}/serviceInfos`)
+            .then(({ data }) => {
+              return new BillingService({
+                ...data,
+                serviceId: data.id,
+                serviceType: DOMAIN_SERVICE_TYPE,
+              });
+            }),
+        );
+        this.trackCancel(DOMAIN_SERVICE_TYPE);
```

Comment:
Do we want the `trackClick` to be fired for each domain selected?

Reply:
yes, each domain deletion must be tracked independently